src/hooks/context/hooks.tsx

```diff
@@ -105,4 +105,52 @@ export const useContextFile = () => {
 			}
 		}
 	}
+}
+
+export interface useSpecFileInput {
+	file?: string,
+	context?: string
+}
+
+export interface useSpecFileOutput {
+	specFile?: SpecificationFile,
+	error?: Error
+}
+
+export const useSpecfile = (flags: useSpecFileInput): useSpecFileOutput => {
+	const contextService: ContextService = container.resolve(ContextService);
+
+	try {
+		let ctx: Context = contextService.loadContextFile();
+		if (flags.file) {
+			let specFile: SpecificationFile = new SpecificationFile(flags.file);
+			if (specFile.isNotValid()) throw new Error("Invalid spec path");
+			return { specFile };
+		}
+
+		if (flags.context) {
+			if (!ctx.store[flags.context]) throw new ContextNotFoundError();
+			//@ts-ignore
+			const specFile = new SpecificationFile(ctx.store[flags.context]);
+			return { specFile };
+		}
```

Comment:
If file flag exists, then loading context before it is unnecessary, so please change it to:

```suggestion
		if (flags.file) {
			const specFile = new SpecificationFile(flags.file);
			if (specFile.isNotValid()) throw new Error("Invalid spec path");
			return { specFile };
		}

              const ctx: Context = contextService.loadContextFile();
		if (flags.context) {
		        const ctxFile = ctx.store[flags.context];
			if (!ctxFile) throw new ContextNotFoundError();
			const specFile = new SpecificationFile(ctxFile);
			return { specFile };
		}
```

Reply:
And change all `let` to `const` in places where you don't change the reference to variable.